covid19-etl/etl/cityofchicago.py

```diff
@@ -0,0 +1,410 @@
+            "hospitalizations_other_race_non_latinx",
+            "hospitalizations_unknown_race_ethnicity",
+        ]
+
+    def get_unified_date_format(self, datetime):
+        """
+        Method to parse and check the date from datasource used in this ETL
+        """
+        date = datetime
+        if "T" in datetime:
+            date = datetime.split("T")[0]
+        year, month, day = date.split("-")
+        # format all the dates the same way
+        if len(year) == 2:
+            year = "20{}".format(year)
+        if len(month) == 1:
+            month = "0{}".format(month)
+        if len(day) == 1:
+            day = "0{}".format(day)
+        return "-".join((year, month, day))
+
+    def convert_str_to_int(self, string):
+        # Method to parse numbers from string to int and 0 if string it empty
+
+        return int(string or 0)
+
+    def get_summary_location(self):
+        # This dataset would only require one `summary_location` which is chicago, so the entry is made in `summary_location` only if it doesn't already exsist
+
+        current_summary_location = self.metadata_helper.get_existing_summary_locations()
+        if len(current_summary_location) > 0:
+            return
+        else:
+            self.summary_locations[self.summary_location_submitter_id] = {
+                "country_region": self.country,
+                "county": self.county,
+                "province_state": self.state,
+                "projects": [{"code": self.project_code}],
+            }
+
+    def add_summary_clinical(
+        self, cases_total, deaths_total, hospitalizations_total, today
+    ):
+
+        # To add `summary_clinical` data for each date in dataset with total of cases, deaths and hospitalization records
+
+        summary_clinical_submitter_id = derived_submitter_id(
+            self.summary_location_submitter_id,
+            "summary_location",
+            "summary_clinical",
+            {"date": today.strftime("%Y-%m-%d")},
+        )
+        summary_clinical = {
+            "submitter_id": summary_clinical_submitter_id,
+            "count": cases_total,
+            "deaths": deaths_total,
+            "hospitaliIzedCumulative": hospitalizations_total,
+            "date": today.strftime("%Y-%m-%dT%H:%M:%S"),
+            "summary_locations": [{"submitter_id": self.summary_location_submitter_id}],
+        }
+        self.summary_clinicals[summary_clinical_submitter_id] = summary_clinical
+        return summary_clinical_submitter_id
+
+    def add_to_summary_group_demographics(
+        self, header_value, record_value, summary_clinical_submitter_id
+    ):
+
+        # To add `summary_group_demographics` data for each date in dataset with total of cases, deaths and hospitalization records
+
+        # `summary_group_demographics` have summary of covid data for either AgeGroup, Race, Gender
+        age_group_dict = {"AgeGroup": "None", "Race": "None", "Gender": "None"}
```

Comment:
for `age_group_dict` and `summary_group_demographics_value_dict`: you don't need to initialize the values since you set them later. `age_group_dict = {}` would work, and if a key is missing, the ETL will fail to submit, which is a better outcome than silently submitting null data

Reply:
ok makes sense, it could be renamed though, it doesn't only contain age group data. maybe submitter_id_dict

and what about the default values in `summary_group_demographics_value_dict`? do we need them?

Reply:
- Yes I can change it to `submitter_id_dict`
- About `summary_group_demographics_value_dict`,  the reason for having it to initialize the dict for `summary_group_demographics` so every time it would have same structure and we can change the needed values

Reply:
for `summary_group_demographics_value_dict` i'll go back to my initial comment then:

> `{}` would work, and if a key is missing, the ETL will fail to submit, which is a better outcome than silently submitting null data

Reply:
I added those to add default value columns `age_group`, `race`, `gender` and `ethnicity` as null in `summary_group_demographics_value_dict` so I don't have to add those in each conditions.

If I can proceed with other key not mentioned in the dict, I can pass the null value which would give me a dict like this,

{'age_group': 'less than 17', 'count': 2, 'deaths': 0, 'hospitalizations': 0, 'summary_clinicals': [{'submitter_id': 'summary_clinical_us_il_chicago_2022-04-29'}]}

which is currently this

{'age_group': 'less than 17', 'race': '', 'gender': '', 'ethnicity': '', 'count': 2, 'deaths': 0, 'hospitalizations': 0, 'summary_clinicals': [{'submitter_id': 'summary_clinical_us_il_chicago_2022-04-29'}] }

Reply:
i think your first example would work?

Reply:
As long as it works while submitting data, I can make the change

Reply:
I added changes as new commit